maven-core/src/main/java/org/apache/maven/plugin/MethodMap.java

```diff
@@ -0,0 +1,395 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.plugin;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Hashtable;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+class MethodMap {
+    private static final int MORE_SPECIFIC = 0;
+
+    private static final int LESS_SPECIFIC = 1;
+
+    private static final int INCOMPARABLE = 2;
+
+    /**
+     * Keep track of all methods with the same name.
+     */
+    Map<String, List<Method>> methodByNameMap = new Hashtable<>();
+
+    /**
+     * Add a method to a list of methods by name. For a particular class we are keeping track of all the methods with
+     * the same name.
+     *
+     * @param method The method
+     */
+    public void add(Method method) {
+        String methodName = method.getName();
+
+        List<Method> l = get(methodName);
```

Comment:
l --> list

In general, avoid abbreviations and especially avoid the single letter L which is confusable with the digit one in many fonts